src/data/sym/card.lean

```diff
@@ -36,9 +35,185 @@ Prove the general case of stars and bars.
 stars and bars
 -/
 
+def multichoose1 (n k : ℕ) := fintype.card (sym (fin n) k)
+
+def multichoose2 (n k : ℕ) := (n + k - 1).choose k
+
+def encode (n k : ℕ) (x : sym (fin n.succ) k.succ) : sym (fin n) k.succ ⊕ sym (fin n.succ) k :=
+if h : fin.last n ∈ x then
+  sum.inr ⟨x.val.erase (fin.last n), by { rw [multiset.card_erase_of_mem h, x.property], refl }⟩
```

Comment:
Can you extract everything to the right of `sum.inr` to a new definition, `sym2.erase`?

Reply:
```suggestion
sum.inr (x.erase (fin.last n))
```